unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -105,6 +105,17 @@ public static void freeMemory(long address) {
     _UNSAFE.freeMemory(address);
   }
 
+  public static long reallocateMemory(long address, long oldSize, long newSize) {
```

Comment:
I have this in the same group of functions as freeMemory, allocateMemory. What would the on heap version look like?